# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/cocoa/GfxInfo.h
# Commit: f4a5301f80a1
# Full Hash: f4a5301f80a1c5111a543081d289022e0d3cfe7f
# Author: Bert Peers <bpeers@mozilla.com>
# Date: 2020-10-03 10:23:35
# Regressor Bug: 1649490
# File Overlap Count: 2
# Description:
#   Bug 1668145 - Detect all Mac GPUs. r=jrmuizel,haik
#   
#   This also adds a diagonstic assert to make sure the code works in all process types.
#   And it adds another item to the sandbox IOKit property name allowlist, so that
#   the detection works in content processes.
# ==============================================================================

diff -r 22be77d681b6 -r f4a5301f80a1 widget/cocoa/GfxInfo.h
--- a/widget/cocoa/GfxInfo.h	Fri Oct 02 15:00:28 2020 +0000
+++ b/widget/cocoa/GfxInfo.h	Fri Oct 02 01:10:20 2020 +0000
@@ -81,14 +81,16 @@
   void GetSelectedCityInfo();
   void AddCrashReportAnnotations();
 
-  uint32_t mAdapterRAM;
-  nsString mDeviceID;
-  nsString mDriverVersion;
-  nsString mDriverDate;
-  nsString mDeviceKey;
+  uint32_t mNumGPUsDetected;
 
-  nsString mAdapterVendorID;
-  nsString mAdapterDeviceID;
+  uint32_t mAdapterRAM[2];
+  nsString mDeviceID[2];
+  nsString mDriverVersion[2];
+  nsString mDriverDate[2];
+  nsString mDeviceKey[2];
+
+  nsString mAdapterVendorID[2];
+  nsString mAdapterDeviceID[2];
 
   uint32_t mOSXVersion;
 };